fhirstore/search/url_parser.py

```diff
@@ -0,0 +1,137 @@
+        self.elements = None
+        self.include = False
+
+        # meta args
+        self.summary = False
+        self.is_summary_count = False
+        self.offset = 0
+        self.result_size = 100
+        self.processed_params = pre_process_params(url_args)
+        self.process_params()
+
+    def sort_params(self):
+        has_sort = None
+        if "_sort" in self.processed_params:
+            has_sort = self.processed_params["_sort"]
+        elif "multiple" in self.processed_params:
+            has_sort = self.processed_params["multiple"].get("_sort", None)
+        # if there is a sorting argument, process it to handle a change of sorting order
+        if has_sort:
+            sorting_params = []
+            for argument in has_sort:
+                # find a "-" before the argument. It indicates a sorting order different from default
+                has_minus = re.search(r"^-(.*)", argument)
+                if has_minus:
+                    # if the argument is -score, sort by ascending order (i.e. ascending relevance)
+                    if has_minus.group(1) == "_score":
+                        sorting_params.append({has_minus.group(1): {"order": "asc"}})
+                    # for any other argument, sort by descending order
+                    else:
+                        sorting_params.append({has_minus.group(1): {"order": "desc"}})
+                # if there is no "-", use order defaults defined in elasticsearch
+                else:
+                    sorting_params.append(argument)
+            self.sort = sorting_params
+
+    def include_params(self):
+        if "_include" in self.processed_params:
+            self.include = [re.search(r"(.*):(.*)", self.processed_params["_include"][0]).group(2)]
+
+        elif (
+            "multiple" in self.processed_params and "_include" in self.processed_params["multiple"]
+        ):
+            attributes = self.processed_params["multiple"].get("_include", None)
+            self.include = [re.search(r"(.*):(.*)", elem).group(2) for elem in attributes]
+
+    # Removes result formatting arguments from the search fields
+    def clean_params(self):
+        if "multiple" in self.processed_params:
+            self.processed_params["multiple"].pop("_element", None)
+            self.processed_params["multiple"].pop("_sort", None)
+            self.processed_params["multiple"].pop("_include", None)
+            self.processed_params["multiple"].pop("_type", None)
+
+        if self.processed_params.get("multiple") == {}:
+            self.processed_params = {}
+
+        self.processed_params.pop("_summary", None)
+        self.processed_params.pop("_element", None)
+        self.processed_params.pop("_sort", None)
+        self.processed_params.pop("_include", None)
+        self.processed_params.pop("_count", None)
+
+        # Probably not the best way to proceed
+        self.core_args = self.processed_params
+
+    def process_params(self):
```

Comment:
what does this do ? "process" is too vague if there's no documentation explaining what this function does